Lookout/lookout_modules/connector_mobile_endpoint_security.py

```diff
@@ -0,0 +1,173 @@
+                    self.retry_ms = ss_event.retry
+
+                self.log(f"{ss_event.event} event received, shutting down...", level="info")
+                if self.retry_ms:
+                    wait_time_seconds = round(self.retry_ms / 1000.0, 2)
+                    self.log(f"Waiting %d seconds before reconnect" % wait_time_seconds)
+
+                    time.sleep(wait_time_seconds)
+
+                break
+
+        # shutdown
+        sse_client.close()
+
+    def run(self) -> None:
+        while self.running:
+            for event in self.fetch_events():
+                if event.event == "events":
+                    mra_events = []
+                    try:
+                        mra_events = json.loads(event.data).get("events", [])
+
+                    except ValueError as e:
+                        logger.error(f"failed to parse mra events from sse client: {e}")
+
+                    if len(mra_events) > 0:
+                        most_recent_date_seen = max(isoparse(item["created_time"]) for item in mra_events)
+                        batch_of_events = [orjson.dumps(event).decode("utf-8") for event in mra_events]
+
+                        self.log(
+                            message=f"Forwarded {len(batch_of_events)} events to the intake",
+                            level="info",
+                        )
+                        OUTCOMING_EVENTS.labels(intake_key=self.connector.configuration.intake_key).inc(
+                            len(batch_of_events)
+                        )
+                        self.connector.push_events_to_intakes(events=batch_of_events)
+
+                        # save last seen event id
+                        self.cursor.offset = self.last_event_id
+
+                        now = datetime.now(timezone.utc)
+                        current_lag = now - most_recent_date_seen
+                        EVENTS_LAG.labels(intake_key=self.connector.configuration.intake_key).set(
+                            current_lag.total_seconds()
+                        )
+
+                elif event.event == "heartbeat":
+                    logger.debug("Received heartbeat")
+                    EVENTS_LAG.labels(intake_key=self.connector.configuration.intake_key).set(0)
+
+
+class MobileEndpointSecurityConnector(Connector):
+    module: LookoutModule
+    configuration: MobileEndpointSecurityConnectorConfiguration
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def run(self) -> None:
+        self.log("Starting collecting Lookout MES events", level="info")
+
+        worker = MobileEndpointSecurityThread(connector=self)
+
+        try:
+            worker.start()
+
+            while self.running:
+                if worker.running and not worker.is_alive():
+                    worker = MobileEndpointSecurityThread(connector=self)
+                    worker.start()
+
+                time.sleep(5)
+
+            worker.stop()
```

Comment:
Why do you use a thread? Cannot the code in the thread run inside the connector directly?